docs/read_and_write.rst

```diff
@@ -108,6 +108,34 @@ to those files. Any subsequent queries will not return the deleted rows.
   Do not read datasets with deleted rows using Lance versions prior to 0.5.0,
   as they will return the deleted rows. This is fixed in 0.5.0 and later.
 
+Updating rows
+~~~~~~~~~~~~~
+
+Lance supports updating rows based on SQL expressions with the
+:py:meth:`lance.LanceDataset.update` method. For example, if we notice
+that Bob's name in our dataset has been sometimes written as ``Blob``, we can fix
+that with:
+
+  .. code-block:: python
+    import lance
+
+    dataset = lance.dataset("./alice_and_bob.lance")
+    dataset.update(dict(name = "'Bob'"), where="name = 'Blob'")
```

Comment:
```suggestion
    dataset.update({"name": "'Bob'"}), where="name = 'Blob'")
```
Would a dictionary literal be more intuitive here?

Reply:
I guess so. I just find writing the quotes exhausting.